jenkins/deploy/Jenkinsfile

```diff
@@ -0,0 +1,96 @@
+pipeline {
+    agent any
+
+    options {
+        buildDiscarder(logRotator(numToKeepStr: "5"))
+        timeout time: 60, unit: "MINUTES"
+    }
+
+    stages {
+        stage("Build") {
+            steps {
+                configFileProvider([configFile(fileId: "iis-build-properties", variable: 'BUILD_PROPERTIES')]) {
+                    load "${BUILD_PROPERTIES}"
+                    withEnv(["JAVA_HOME=${ tool type: 'jdk', name: "${JDK_VERSION}" }",
+                             "PATH+MAVEN=${tool type: 'maven', name: "${MAVEN_VERSION}"}/bin:${env.JAVA_HOME}/bin"]) {
+                        sh """
+                            mvn clean compile
+                        """
+                    }
+                }
+            }
+        }
+
+        stage("Test") {
+            steps {
+                configFileProvider([configFile(fileId: "iis-build-properties", variable: 'BUILD_PROPERTIES')]) {
+                    load "${BUILD_PROPERTIES}"
+                    withEnv(["JAVA_HOME=${ tool type: 'jdk', name: "$JDK_VERSION" }",
+                             "PATH+MAVEN=${tool type: 'maven', name: "$MAVEN_VERSION"}/bin:${env.JAVA_HOME}/bin"]) {
+                        sh """
+                            mvn test
+                        """
+                    }
+                }
+            }
+        }
+
+        stage("Install") {
+            steps {
+                configFileProvider([configFile(fileId: "iis-build-properties", variable: 'BUILD_PROPERTIES')]) {
+                    load "${BUILD_PROPERTIES}"
+                    withEnv(["JAVA_HOME=${ tool type: 'jdk', name: "${JDK_VERSION}" }",
+                             "PATH+MAVEN=${tool type: 'maven', name: "${MAVEN_VERSION}"}/bin:${env.JAVA_HOME}/bin"]) {
+                        sh """
+                            mvn clean install -DskipTests
+                        """
+                    }
+                }
+            }
+        }
+
+        stage("Upload") {
+            steps {
+                configFileProvider([configFile(fileId: "iis-build-properties", variable: 'BUILD_PROPERTIES')]) {
+                    load "${BUILD_PROPERTIES}"
+                    withEnv(["JAVA_HOME=${ tool type: 'jdk', name: "${JDK_VERSION}" }",
+                             "PATH+MAVEN=${tool type: 'maven', name: "${MAVEN_VERSION}"}/bin:${env.JAVA_HOME}/bin"]) {
+                        sshagent(['iis-deployment-credentials']) {
+                            sh """
+                                cd iis-wf/iis-wf-primary && \
+                                mvn clean package -Poozie-package,deploy \
+                                    -Dworkflow.source.dir=eu/dnetlib/iis/wf/primary/main \
+                                    -DiisConnectionProperties=${WORKSPACE}/connection-properties/iis-cdh5-test-gw.properties
+                            """
+                        }
+                    }
+                }
+            }
+        }
+
+        stage("Deploy") {
```

Comment:
Yes, that's a good idea, I've renamed the stages to `Deploy` and `Release`.

Reply:
Actually I've renamed the stages to `Deploy Primary` and `Release Primary` because these stages deal with primary workflow.  For deployment of other workflows we should have other Jenkinsfile or add stages to existing file.